.github/workflows/macos.yml

```diff
@@ -9,6 +9,13 @@ on:
 jobs:
   general:
     runs-on: macOS-latest
+
+    strategy:
+      matrix:
+        build-tool: [autotools, cmake]
+        cc: [gcc, clang]
```

Comment:
Why do we want to test with gcc? `gcc` is not really a native component of macOS systems, but we use gcc everywhere else in our CI.

Reply:
gcc is a must have now, clang has several issues
especially with C++

Reply:
Maybe I'm wrong, but I can't see more than 10 packages depending on gcc.

https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-core+gcc&type=code&p=1

Reply:
@MrAnno Do you have any idea why CMake does not pick up and use the CFLAGS environment variable
the build log shows nicely that the env is set correctly
```
Run cmake --build ./build -j ${THREADS} --target install --verbose
  cmake --build ./build -j ${THREADS} --target install --verbose
  shell: /bin/bash -e {0}
  env:
    HOMEBREW_PREFIX: /usr/local
    PYTHONUSERBASE: /Users/runner/python_packages
    CC: gcc
    PKG_CONFIG_PATH: /usr/local/opt/openssl@3/lib/pkgconfig:/usr/local/opt/net-snmp/lib/pkgconfig:
    THREADS: 3
    CONFIGURE_FLAGS: --enable-cpp --prefix= --enable-all-modules --enable-tests --with-ivykis=system --with-python=3 --disable-smtp --disable-grpc --disable-java --disable-java-modules --disable-mqtt --disable-pacct
    CFLAGS: -I/usr/local/include/
    LDFLAGS: -L/usr/local/lib
    CMAKE_FLAGS: -DENABLE_CPP=ON -DCMAKE_BUILD_TYPE=Debug -DSUMMARY_FULL=ON -DBUILD_TESTING=OFF -DCMAKE_C_FLAGS=-Wall -DIVYKIS_SOURCE=system -DPYTHON_VERSION=3 -DENABLE_AFSMTP=OFF -DENABLE_GRPC=OFF -DENABLE_JAVA=OFF -DENABLE_JAVA_MODULES=OFF -DENABLE_MQTT=OFF -DENABLE_PACCT=OFF
```
but in the `--verbose` CMake build logs it can be seen that the content of the CFLAGS (-I/usr/local/include/) is not appended to the compile command param list, e.g.

```
cd /Users/runner/work/syslog-ng/syslog-ng/build/modules/afsql && /usr/local/bin/gcc -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_32 -DYYDEBUG=1 -D_GNU_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -D__APPLE_USE_RFC_3542 -Dafsql_EXPORTS -I/Users/runner/work/syslog-ng/syslog-ng/build -I/Users/runner/work/syslog-ng/syslog-ng/modules/afsql -I/Users/runner/work/syslog-ng/syslog-ng/build/modules/afsql -I/Users/runner/work/syslog-ng/syslog-ng/lib -I/Users/runner/work/syslog-ng/syslog-ng/build/lib -isystem /usr/local/include/dbi -isystem /usr/local/Cellar/openssl@3/3.1.2/include -isystem /Library/Frameworks/Mono.framework/Headers -isystem /usr/local/Cellar/ivykis/0.42.4/include -isystem /usr/local/Cellar/pcre2/10.42/include -isystem /Users/runner/work/syslog-ng/syslog-ng/lib/eventlog/src -isystem /usr/local/Cellar/glib/2.76.4/include/glib-2.0 -isystem /usr/local/Cellar/glib/2.76.4/lib/glib-2.0/include -Wall -g -std=gnu99 -isysroot /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk -mmacosx-version-min=12.6 -fPIC -Wshadow -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros -Wold-style-declaration -Wunused-but-set-parameter -Woverride-init -Wimplicit-function-declaration -Wnested-externs -Wstrict-prototypes -Wswitch-default -Wimplicit-int -Wall -Wuninitialized -Wdeprecated -Wdeprecated-declarations -Woverflow -Wdouble-promotion -Wfloat-equal -Wpointer-arith -Wpointer-sign -Wmissing-format-attribute -Wold-style-definition -Wundef -Wignored-qualifiers -Wfloat-conversion -Wbad-function-cast -MD -MT modules/afsql/CMakeFiles/afsql.dir/afsql-grammar.c.o -MF CMakeFiles/afsql.dir/afsql-grammar.c.o.d -o CMakeFiles/afsql.dir/afsql-grammar.c.o -c /Users/runner/work/syslog-ng/syslog-ng/build/modules/afsql/afsql-grammar.c
/Users/runner/work/syslog-ng/syslog-ng/modules/afsocket/afinet-dest.c:46:10: fatal error: libnet.h: No such file or directory
   46 | #include <libnet.h>
      |          ^~~~~~~~~~
compilation terminated.
```
that leads to an error of course.
What could cause this?

Reply:
The reasons why running gcc tests might be a good idea too are

- now only gcc can compile the C++ codes and it is turned on  for it now, C++ compilation differences can also occure
- there were many differences between the gcc and clang versions
- several times too late it turned out the cmake configs did not altered properly to run fine on macOS as well

So, I think the whole matrix test would be super useful here and can prevent many later annoyances.

(btw, it would be best if we could run the tests for ARM, Silicon macs too, but unfortunately, there's no such machine yet in the GitLab CI :/ )